includes/embeds/class-amp-gutenberg-categories-handler.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Class AMP_Gutenberg_Categories_Handler
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Gutenberg_Categories_Handler
+ *
+ * @since 1.0
+ */
+class AMP_Gutenberg_Categories_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Register embed.
+	 */
+	public function register_embed() {
+		if ( function_exists( 'gutenberg_init' ) ) {
+			add_action( 'the_post', array( $this, 'override_category_block_render_callback' ) );
```

Comment:
I don't think this needs to wait for `the_post` action. I think it can just do `$this-> override_category_block_render_callback()`